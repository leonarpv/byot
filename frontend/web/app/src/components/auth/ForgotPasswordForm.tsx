import * as React from 'react';
import {Form, Formik} from 'formik';
import {Input} from '@byot-frontend/web-common/src/components/elementary/form/Input';
import {TFunction} from 'i18next';
import * as Yup from 'yup';
import {useTranslation} from 'react-i18next';
import {PlainLayoutNarrow} from '@byot-frontend/web-common/src/components/plain-layout/PlainLayoutNarrow';
import {PlainLayoutTitle} from '@byot-frontend/web-common/src/components/plain-layout/PlainLayoutTitle';
import {Button} from '@byot-frontend/web-common/src/components/elementary/form/Button';

const loginSchema = (t: TFunction) =>
  Yup.object().shape({
    email: Yup.string().required(t('Enter email')).email(t('Enter valid email')),
  });

interface Props {
  onSubmit: (email: string) => void;
  loading?: boolean;
}

export const ForgotPasswordForm: React.FC<Props> = (props: Props) => {
  const initialValues = {email: ''};
  const {t} = useTranslation();
  return (
    <PlainLayoutNarrow>
      <PlainLayoutTitle>{t('Reset password')}</PlainLayoutTitle>
      <Formik
        initialValues={initialValues}
        validateOnBlur
        validateOnChange
        validationSchema={loginSchema(t)}
        onSubmit={({email}) => props.onSubmit(email)}>
        {({values, handleChange, handleBlur, errors, touched}) => (
          <Form data-testid="common-auth-forgotPassword-form-form">
            <Input
              data-testid="common-auth-forgotPassword-form-email"
              label={t('Enter email')}
              errorText={touched.email && errors.email}
              onChange={handleChange('email')}
              onBlur={handleBlur('email')}
              value={values.email}
            />

            <Button type="submit" color="gradient" fullWidth loading={props.loading}>
              {t('Request reset password')}
            </Button>
          </Form>
        )}
      </Formik>
    </PlainLayoutNarrow>
  );
};
