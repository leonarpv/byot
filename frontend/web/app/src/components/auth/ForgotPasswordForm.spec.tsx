import * as React from 'react';
import {ForgotPasswordForm} from './ForgotPasswordForm';
import {fireEvent, render, waitFor} from '@testing-library/react';

describe('<ForgotPasswordForm/>', () => {
  it('should successfully request reset', async () => {
    const onSubmit = jest.fn();
    const wrapper = render(<ForgotPasswordForm onSubmit={onSubmit} />);

    const emailValue = 'john@doe.com';
    const email = wrapper!.queryByTestId('common-auth-forgotPassword-form-email')!.querySelector('input')!;
    expect(email).not.toBeNull();
    fireEvent.change(email, {target: {value: emailValue}});
    expect(email.value).toEqual(emailValue);
    const form = wrapper!.queryByTestId('common-auth-forgotPassword-form-form')!;
    expect(form).not.toBeNull();
    fireEvent.submit(form);
    await waitFor(() => expect(onSubmit).toBeCalledWith(emailValue));
    expect(wrapper).toMatchSnapshot();
  });
  it('should fail requesting', async () => {
    const onSubmit = jest.fn();
    const wrapper = render(<ForgotPasswordForm onSubmit={onSubmit} />);

    const emailValue = 'john@doe';
    const email = wrapper!.queryByTestId('common-auth-forgotPassword-form-email')!.querySelector('input')!;
    expect(email).not.toBeNull();
    fireEvent.change(email, {target: {value: emailValue}});
    expect(email.value).toEqual(emailValue);
    const form = wrapper!.queryByTestId('common-auth-forgotPassword-form-form')!;
    expect(form).not.toBeNull();
    fireEvent.submit(form);
    await waitFor(() => expect(onSubmit).not.toBeCalled());
    expect(wrapper).toMatchSnapshot();
  });
});
