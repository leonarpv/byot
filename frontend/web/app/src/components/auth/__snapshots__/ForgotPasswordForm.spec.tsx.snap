// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForgotPasswordForm/> should fail requesting 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="makeStyles-root-11"
      >
        <div
          class="MuiBox-root MuiBox-root-12"
        >
          <h3
            class="MuiTypography-root MuiTypography-h3"
          >
            Reset password
          </h3>
        </div>
        <form
          action="#"
          data-testid="common-auth-forgotPassword-form-form"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <div
                class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                data-testid="common-auth-forgotPassword-form-email"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root makeStyles-label-14 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled Mui-error Mui-error MuiFormLabel-filled"
                  data-shrink="true"
                >
                  Enter email
                </label>
                <div
                  class="MuiInputBase-root MuiFilledInput-root makeStyles-filledRoot-13 makeStyles-filledRoot-15 Mui-error Mui-error MuiInputBase-fullWidth MuiInputBase-formControl"
                >
                  <input
                    aria-invalid="true"
                    class="MuiInputBase-input MuiFilledInput-input"
                    type="text"
                    value="john@doe"
                  />
                </div>
              </div>
              <div
                class="MuiCollapse-container"
                style="min-height: 0px; height: 0px; transition-duration: 300ms;"
              >
                <div
                  class="MuiCollapse-wrapper"
                >
                  <div
                    class="MuiCollapse-wrapperInner"
                  >
                    <p
                      class="MuiFormHelperText-root MuiFormHelperText-contained Mui-error"
                    >
                      Enter valid email
                    </p>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <button
                class="MuiButtonBase-root MuiButton-root makeStyles-root-16 makeStyles-root-19 MuiButton-contained MuiButton-containedPrimary MuiButton-disableElevation MuiButton-fullWidth"
                tabindex="0"
                type="submit"
              >
                <span
                  class="MuiButton-label"
                >
                  <span
                    class="makeStyles-inner-17 makeStyles-inner-20"
                    data-testid="common-elementary-button-children"
                    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
                  >
                    Request reset password
                  </span>
                  <div
                    class="makeStyles-loader-18"
                    data-testid="common-elementary-button-loader"
                    style="opacity: 0; visibility: hidden;"
                  >
                    <div
                      class="css-0"
                    >
                      <div
                        class="css-1ab95gp"
                      />
                      <div
                        class="css-tm4xur"
                      />
                      <div
                        class="css-1epl0i6"
                      />
                      <div
                        class="css-1fj2adh"
                      />
                      <div
                        class="css-1vivfwt"
                      />
                    </div>
                  </div>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="makeStyles-root-11"
    >
      <div
        class="MuiBox-root MuiBox-root-12"
      >
        <h3
          class="MuiTypography-root MuiTypography-h3"
        >
          Reset password
        </h3>
      </div>
      <form
        action="#"
        data-testid="common-auth-forgotPassword-form-form"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
              data-testid="common-auth-forgotPassword-form-email"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root makeStyles-label-14 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled Mui-error Mui-error MuiFormLabel-filled"
                data-shrink="true"
              >
                Enter email
              </label>
              <div
                class="MuiInputBase-root MuiFilledInput-root makeStyles-filledRoot-13 makeStyles-filledRoot-15 Mui-error Mui-error MuiInputBase-fullWidth MuiInputBase-formControl"
              >
                <input
                  aria-invalid="true"
                  class="MuiInputBase-input MuiFilledInput-input"
                  type="text"
                  value="john@doe"
                />
              </div>
            </div>
            <div
              class="MuiCollapse-container"
              style="min-height: 0px; height: 0px; transition-duration: 300ms;"
            >
              <div
                class="MuiCollapse-wrapper"
              >
                <div
                  class="MuiCollapse-wrapperInner"
                >
                  <p
                    class="MuiFormHelperText-root MuiFormHelperText-contained Mui-error"
                  >
                    Enter valid email
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <button
              class="MuiButtonBase-root MuiButton-root makeStyles-root-16 makeStyles-root-19 MuiButton-contained MuiButton-containedPrimary MuiButton-disableElevation MuiButton-fullWidth"
              tabindex="0"
              type="submit"
            >
              <span
                class="MuiButton-label"
              >
                <span
                  class="makeStyles-inner-17 makeStyles-inner-20"
                  data-testid="common-elementary-button-children"
                  style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
                >
                  Request reset password
                </span>
                <div
                  class="makeStyles-loader-18"
                  data-testid="common-elementary-button-loader"
                  style="opacity: 0; visibility: hidden;"
                >
                  <div
                    class="css-0"
                  >
                    <div
                      class="css-1ab95gp"
                    />
                    <div
                      class="css-tm4xur"
                    />
                    <div
                      class="css-1epl0i6"
                    />
                    <div
                      class="css-1fj2adh"
                    />
                    <div
                      class="css-1vivfwt"
                    />
                  </div>
                </div>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<ForgotPasswordForm/> should successfully request reset 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="makeStyles-root-1"
      >
        <div
          class="MuiBox-root MuiBox-root-2"
        >
          <h3
            class="MuiTypography-root MuiTypography-h3"
          >
            Reset password
          </h3>
        </div>
        <form
          action="#"
          data-testid="common-auth-forgotPassword-form-form"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <div
                class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                data-testid="common-auth-forgotPassword-form-email"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root makeStyles-label-4 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled MuiFormLabel-filled"
                  data-shrink="true"
                >
                  Enter email
                </label>
                <div
                  class="MuiInputBase-root MuiFilledInput-root makeStyles-filledRoot-3 makeStyles-filledRoot-5 MuiInputBase-fullWidth MuiInputBase-formControl"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiFilledInput-input"
                    type="text"
                    value="john@doe.com"
                  />
                </div>
              </div>
              <div
                class="MuiCollapse-container MuiCollapse-hidden"
                style="min-height: 0px;"
              >
                <div
                  class="MuiCollapse-wrapper"
                >
                  <div
                    class="MuiCollapse-wrapperInner"
                  >
                    <p
                      class="MuiFormHelperText-root MuiFormHelperText-contained"
                    >
                      <span>
                        ​
                      </span>
                    </p>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <button
                class="MuiButtonBase-root MuiButton-root makeStyles-root-6 makeStyles-root-9 MuiButton-contained MuiButton-containedPrimary MuiButton-disableElevation MuiButton-fullWidth"
                tabindex="0"
                type="submit"
              >
                <span
                  class="MuiButton-label"
                >
                  <span
                    class="makeStyles-inner-7 makeStyles-inner-10"
                    data-testid="common-elementary-button-children"
                    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
                  >
                    Request reset password
                  </span>
                  <div
                    class="makeStyles-loader-8"
                    data-testid="common-elementary-button-loader"
                    style="opacity: 0; visibility: hidden;"
                  >
                    <div
                      class="css-0"
                    >
                      <div
                        class="css-1ab95gp"
                      />
                      <div
                        class="css-tm4xur"
                      />
                      <div
                        class="css-1epl0i6"
                      />
                      <div
                        class="css-1fj2adh"
                      />
                      <div
                        class="css-1vivfwt"
                      />
                    </div>
                  </div>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="makeStyles-root-1"
    >
      <div
        class="MuiBox-root MuiBox-root-2"
      >
        <h3
          class="MuiTypography-root MuiTypography-h3"
        >
          Reset password
        </h3>
      </div>
      <form
        action="#"
        data-testid="common-auth-forgotPassword-form-form"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
              data-testid="common-auth-forgotPassword-form-email"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root makeStyles-label-4 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled MuiFormLabel-filled"
                data-shrink="true"
              >
                Enter email
              </label>
              <div
                class="MuiInputBase-root MuiFilledInput-root makeStyles-filledRoot-3 makeStyles-filledRoot-5 MuiInputBase-fullWidth MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiFilledInput-input"
                  type="text"
                  value="john@doe.com"
                />
              </div>
            </div>
            <div
              class="MuiCollapse-container MuiCollapse-hidden"
              style="min-height: 0px;"
            >
              <div
                class="MuiCollapse-wrapper"
              >
                <div
                  class="MuiCollapse-wrapperInner"
                >
                  <p
                    class="MuiFormHelperText-root MuiFormHelperText-contained"
                  >
                    <span>
                      ​
                    </span>
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <button
              class="MuiButtonBase-root MuiButton-root makeStyles-root-6 makeStyles-root-9 MuiButton-contained MuiButton-containedPrimary MuiButton-disableElevation MuiButton-fullWidth"
              tabindex="0"
              type="submit"
            >
              <span
                class="MuiButton-label"
              >
                <span
                  class="makeStyles-inner-7 makeStyles-inner-10"
                  data-testid="common-elementary-button-children"
                  style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
                >
                  Request reset password
                </span>
                <div
                  class="makeStyles-loader-8"
                  data-testid="common-elementary-button-loader"
                  style="opacity: 0; visibility: hidden;"
                >
                  <div
                    class="css-0"
                  >
                    <div
                      class="css-1ab95gp"
                    />
                    <div
                      class="css-tm4xur"
                    />
                    <div
                      class="css-1epl0i6"
                    />
                    <div
                      class="css-1fj2adh"
                    />
                    <div
                      class="css-1vivfwt"
                    />
                  </div>
                </div>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
